utils/python/CIME/compare_test_results.py

```diff
@@ -60,39 +67,74 @@ def compare_test_results(baseline_name, baseline_root, test_root, compiler, test
+                    compare_comment = "Test did not make it to run phase"
+                    do_compare = False
+                elif (run_result != TEST_PASS_STATUS):
+                    compare_result = "SKIP"
+                    compare_comment = "Run phase did not pass"
+                    do_compare = False
                 else:
-                    compare_result = TEST_FAIL_STATUS
-                    all_pass_or_skip = False
-
-                # Following the logic in SystemTestsCommon._compare_baseline:
-                # We'll print the comment if it's a brief one-liner; otherwise
-                # the comment will only appear in the log file
-                if "\n" not in detailed_comments:
-                    compare_comment = detailed_comments
+                    do_compare = True
+            else:
+                do_compare = False
 
-            brief_result = "%s %s %s"%(compare_result, test_name, BASELINE_PHASE)
-            if compare_comment:
-                brief_result += " %s"%(compare_comment)
-            print(brief_result)
+            if nl_do_compare or do_compare:
+                with Case(test_dir) as case:
+                    if nl_do_compare:
+                        nl_success = compare_namelists(case, baseline_name, baseline_root, logfile_name)
+                        if nl_success:
+                            nl_compare_result = TEST_PASS_STATUS
+                            nl_compare_comment = ""
+                        else:
+                            nl_compare_result = TEST_FAIL_STATUS
+                            nl_compare_comment = "See %s/%s" % (test_dir, logfile_name)
+                            all_pass_or_skip = False
+
+                    if do_compare:
+                        success, detailed_comments = compare_history(case, baseline_name, baseline_root, log_id)
+                        if success:
+                            compare_result = TEST_PASS_STATUS
+                        else:
+                            compare_result = TEST_FAIL_STATUS
+                            all_pass_or_skip = False
+
+                        # Following the logic in SystemTestsCommon._compare_baseline:
+                        # We'll print the comment if it's a brief one-liner; otherwise
+                        # the comment will only appear in the log file
+                        if "\n" not in detailed_comments:
+                            compare_comment = detailed_comments
+
+            brief_result = ""
+            if not hist_only:
+                brief_result += "%s %s %s %s\n" % (nl_compare_result, test_name, NAMELIST_PHASE, nl_compare_comment)
+
+            if not namelists_only:
+                brief_result += "%s %s %s" % (compare_result, test_name, BASELINE_PHASE)
+                if compare_comment:
+                    brief_result += " %s" % compare_comment
+                brief_result += "\n"
+
+            print brief_result,
```

Comment:
Okay, that's fine... this probably warrants a broader discussion at some point: Personally, I feel that we should try to avoid writing new code that will obviously break under py3, in order to make our lives easier later... because presumably we will want this to work with py3 at some point in the near future(?). But I don't have the energy to push this point for now :-)